libnavui-dropin/src/main/java/com/mapbox/navigation/dropin/component/infopanel/InfoPanelHeaderComponent.kt

```diff
@@ -39,22 +44,54 @@ internal class InfoPanelHeaderComponent(
             binding.arrivedText.isVisible = it == NavigationState.Arrival
         }
 
-        destinationState.observe {
+        destinationViewModel.state.observe {
             val placeName = it.destination?.features?.firstOrNull()?.placeName()
-            binding.poiName.text =
-                placeName ?: resources.getString(R.string.mapbox_drop_in_dropped_pin)
+            binding.poiName.text = placeName
+                ?: resources.getString(R.string.mapbox_drop_in_dropped_pin)
         }
 
         binding.routePreview.setOnClickListener {
-            dispatch(RoutesAction.FetchAndSetRoute)
+            updateNavigationStateWhenRouteIsReady(NavigationState.RoutePreview)
         }
 
         binding.startNavigation.setOnClickListener {
-            dispatch(RoutesAction.StartNavigation)
+            updateNavigationStateWhenRouteIsReady(NavigationState.ActiveNavigation)
         }
 
         binding.endNavigation.setOnClickListener {
-            dispatch(RoutesAction.StopNavigation)
+            routesViewModel.invoke(RoutesAction.SetRoutes(emptyList()))
+            destinationViewModel.invoke(DestinationAction.SetDestination(null))
+            navigationStateViewModel.invoke(
+                NavigationStateAction.Update(NavigationState.FreeDrive)
+            )
         }
     }
+
+    private fun updateNavigationStateWhenRouteIsReady(navigationState: NavigationState) {
+        val lastPoint = locationViewModel.lastPoint
+            ?: error("Cannot fetch routes when current location is unknown")
```

Comment:
I don't think we would want the app to crash. We can just log it for now, but will have to revisit with the design on how to handle these use cases

Reply:
There is no strong type binding to make sure the NavigationState includes the required values. 

For example, we `NavigationState.RoutePreview` should always have a route. That makes me think we should just add it to the `NavitationState.RoutePreview`.. But then the state is duplicated in the view models 🤔 . 

Will turn it into a warning. But it is important to make it so this state is impossible

Reply:
Yeah definitely these use cases need to be defined. Warning for now should be good.

Reply:
👍